### ISSUE TITLE
Improve logging

### ISSUE THREAD

**Issue by schmunk42** [yiisoft/yii2-swiftmailer#37]
### What steps will reproduce the problem?

- try to send a mail to a non-resolvable host.

### What's expected?

- logs should show an error or warning

```
2016-11-18 05:06:04 [-][-][-][info][yii\mail\BaseMailer::send] Sending email „Willkommen auf My Application" to "idasdjos@hxxxxxxxxn.de"
2016-11-18 05:06:04 [-][-][-][info][yii\swiftmailer\Logger::add] ++ Starting Swift_SmtpTransport
2016-11-18 05:06:04 [-][-][-][warning][yii\swiftmailer\Logger::add] !! Connection could not be established with host mailcatcher [php_network_getaddresses: getaddrinfo failed: Name or service not known #0] (code: 0)
```

### What do you get instead?

- all log messages from mailer in info

```
2016-11-18 05:06:04 [-][-][-][info][yii\mail\BaseMailer::send] Sending email „Willkommen auf My Application" to "idasdjos@hxxxxxxxxn.de"
2016-11-18 05:06:04 [-][-][-][info][yii\swiftmailer\Logger::add] ++ Starting Swift_SmtpTransport
2016-11-18 05:06:04 [-][-][-][info][yii\swiftmailer\Logger::add] !! Connection could not be established with host mailcatcher [php_network_getaddresses: getaddrinfo failed: Name or service not known #0] (code: 0)
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10 (extension: 2.0.6)
| PHP version      | 7.0.12
| Operating system | Docker PHP (Debian)

**Comment by dynasource** [comment]
makes sense to me. I would expect an `error` as the expected output is not realized.

**Comment by klimov-paul** [comment]
Unfortunaly this is not possible - Swiftmailer library does not provide such ability.
Only the log message is avalbale for logging.
See:
https://github.com/yiisoft/yii2-swiftmailer/blob/master/Logger.php#L38

**Comment by schmunk42** [comment]
Can't that be taken from `++` or `!!` - I did not find specs of the logs format, but that looked like  "signs for categories" to me

**Comment by klimov-paul** [comment]
Can't find any explicit documentation on this matter.
All we can use is `LoggerPlugin` source:

https://github.com/swiftmailer/swiftmailer/blob/5.x/lib/classes/Swift/Plugins/LoggerPlugin.php

Searching for particular symbols does not looks reliable to me in a long short, but alright.

**Comment by schmunk42** [comment]
> Searching for particular symbols does not looks reliable to me in a long short, but alright.

I totally agree! But as you said, in this special case, and maybe just looking for `!!` and log a warning - no more magic :)

**Comment by klimov-paul** [comment]
Resolved by commit cd5b231d0a7a3518b1a938b9e24335004f62b9f8
